src/core/io/src/4C_io_mesh.hpp

```diff
@@ -67,97 +82,171 @@ namespace Core::IO::MeshInput
      * cell-block is required to have the same cell-type. 4C can solve different equations on each
      * block.
      */
-    std::map<int, CellBlock> cell_blocks;
-
-    /**
-     * The sides in the mesh. The keys are the side-set IDs, and the values are the side-sets.
-     */
-    std::map<int, SideSet> side_sets;
-
-    /**
-     * The lines in the mesh. The keys are the line-set IDs, and the values are the line-sets.
-     */
-    std::map<int, LineSet> line_sets;
+    std::map<int, CellBlock> cell_blocks{};
 
     /**
      * The points in the mesh. The keys are the point-set IDs, and the values are the point-sets.
      */
-    std::map<int, PointSet> point_sets;
+    std::map<int, PointSet> point_sets{};
   };
 
+  /*!
+   * A cell in a cell-block.
+   */
+  struct CellView
+  {
+    /// External ID of the cell as defined in the mesh
+    int external_id{};
+
+    /// Connectivity of the cell (list of point IDs)
+    std::span<const int> connectivity{};
+  };
 
   /**
    * A cell-block. This encodes a collection of cells of the same type.
    */
-  struct CellBlock
+  class CellBlock
   {
+   public:
     /**
      * The type of the cells in the cell block.
      */
     FE::CellType cell_type;
 
     /**
-     * Cells in this block. The keys are the cell-IDs, and the values are the IDs of the points
-     * making up the cell. The ordering of the points are important for the cell connectivity.
+     * An optional name for the cell block.
+     *
+     * @note Not every file formats provides std::string-names for cell blocks.
+     */
+    std::optional<std::string> name{};
+
+    CellBlock(FE::CellType cell_type) : cell_type(cell_type) {}
+
+    /*!
+     * @brief Returns the number of cells in this block
+     */
+    [[nodiscard]] std::size_t size() const
+    {
+      return cells_connectivity_.size() / FE::num_nodes(cell_type);
+    }
+
+    /*!
+     * @brief Add a cell to this block
+     */
+    void add_cell(int external_id, std::span<const int> connectivity)
+    {
+      FOUR_C_ASSERT_ALWAYS(
+          connectivity.size() == static_cast<std::size_t>(FE::num_nodes(cell_type)),
+          "You are adding a cell with {} points to a cell-block of type {} expecting {} points per "
+          "cell.",
+          connectivity.size(), FE::cell_type_to_string(cell_type), FE::num_nodes(cell_type));
+
+      cells_connectivity_.insert(
+          cells_connectivity_.end(), connectivity.begin(), connectivity.end());
+      external_ids_.push_back(external_id);
+    }
+
+    /*!
+     * @brief Returns a range for iterating over the cell connectivities in this block
+     */
+    [[nodiscard]] auto cell_connectivities() const
+    {
+      auto indices = std::views::iota(std::size_t(0), size());
+      return indices | std::views::transform(
```

Comment:
The simplicity of this is appealing. I know you also suggested this construct to me instead of my recently introduced `NodeRef` and `ElementRef` together with `IteratorRange`. I think it makes sense to go for the std ranges in this PR, since we only use this in internal code, right? For the Discretization, I believe the extra types make things more readable. No idea, if there is a performance difference. We can continue monitoring this as we rework more Discretization internals 👍